src/cim/components/modals/AddBmcHostYamlForm.tsx

```diff
@@ -0,0 +1,200 @@
+  const [showOpenFileButton, setShowOpenFileButton] = React.useState(true);
+  const [showOpeningMessage, setShowOpeningMessage] = React.useState(false);
+  const [yamlContent, setYamlContent] = React.useState<unknown>();
+  const [error, setError] = React.useState<string | undefined>();
+
+  const { t } = useTranslation();
+
+  const downloadYaml = () => {
+    const yaml = [credentials, host1, host2].map((doc) => jsYaml.dump(doc)).join('---\n');
+    const blob = new Blob([yaml], { type: 'text/plain;charset=utf-8' });
+    saveAs(blob, 'bmc_hosts_template.yaml');
+  };
+
+  const importBMAsYAML = () => {
+    const input = document.createElement('input');
+    input.setAttribute('id', 'importBMAsYaml');
+    input.type = 'file';
+    input.accept = '.yaml';
+    input.style.visibility = 'hidden';
+    const body = document.querySelector('body');
+    body?.appendChild(input);
+    input.onchange = (e: Event) => {
+      setShowOpeningMessage(true);
+      input.remove();
+      const target = e.target as HTMLInputElement;
+      const file: File = (target.files as FileList)[0];
+      const maxSize = 12582912; //12 Mb
+      if (file.size > maxSize) {
+        setShowOpenFileButton(false);
+        setFileName(file.name);
+        setFileError(t('ai:The file is too big. Upload a file up to 12 Mb.'));
+        setYamlContent(undefined);
+      } else {
+        const reader = new FileReader();
+        reader.readAsText(file, 'UTF-8');
+        reader.onload = (readerEvent) => {
+          const content = readerEvent.target?.result;
+          // parse yaml
+          if (typeof content === 'string') {
+            try {
+              //await new Promise((r) => setTimeout(r, 100));
+              const yamlContent = jsYaml.loadAll(content);
+              setFileName(file.name);
+              setFileError(undefined);
+              setYamlContent(yamlContent);
+              setShowOpenFileButton(false);
+            } catch (err) {
+              setFileName(file.name);
+              setFileError(
+                t(
+                  'ai:File is not structured correctly. Use the template to use the right file structure.',
+                ),
+              );
+              setYamlContent(undefined);
+              setShowOpenFileButton(false);
+            }
+          }
+        };
+      }
+    };
+    input.click();
+  };
+
+  const deleteFileContent = () => {
+    setFileError(undefined);
+    setFileName(undefined);
+    setShowOpenFileButton(true);
+    setShowOpeningMessage(false);
+  };
+
+  const handleSubmit = async () => {
+    try {
+      setShowOpeningMessage(false);
+      setError(undefined);
+      if (yamlContent !== undefined) {
+        await onCreateBmcByYaml(yamlContent);
+      }
+
+      onClose();
+    } catch (e) {
+      setError(getErrorMessage(e));
+    }
+  };
+
+  return (
+    <Form>
+      <>
+        <ModalBoxBody>
+          <Stack hasGutter>
```

Comment:
If we want the Alert to be outside of the EmptyState, the EmptyState needs to be also wrapped in StackItem. Alternatively we can put the Alert inside EmptyState and use Stack to space the buttons and alerts.